tools/gdb_commands/wt_debug_script_update.py

```diff
@@ -0,0 +1,150 @@
+    return dhandle['handle'].reinterpret_cast(btree).dereference()
+
+def dump_update_chain(update_chain):
+    while True:
+        if not update_chain:
+            print('  λ')
+            break
+        #dbg('update', update_chain)
+        wt_val = update_chain.dereference()
+        obj = None
+        #dbg('wt_val', wt_val)
+        val_bytes = gdb.selected_inferior().read_memory(wt_val['data'], wt_val['size'])
+        can_bson = wt_val['type'] == 3
+        if can_bson:
+            try:
+                obj = bson.decode_all(val_bytes)[0]
+            except:
+                pass
+        print('  ' + '\n  '.join(str(wt_val).split('\n')) + " " + str(obj) + " =>")
+
+        update_chain = update_chain['next']
+
+def dump_insert_list(wt_insert):
+    key_struct = wt_insert['u']['key']
+    key = gdb.selected_inferior().read_memory(int(wt_insert.address) + key_struct['offset'], key_struct['size']).tobytes()
+    print('Key: ' + str(key))
+    print('Value:')
+    update_chain = wt_insert['upd']
+    dump_update_chain(update_chain)
+
+def dump_skip_list(wt_insert_head):
+    wt_insert = wt_insert_head['head'][0]
+    idx = 0
+    while True:
+        if not wt_insert:
+            break
+        dump_insert_list(wt_insert.dereference())
+        #dbg('insert' + str(idx), wt_insert.dereference())
+        idx+=1
+        wt_insert = wt_insert['next'][0]
+
+def dump_modified(leaf_page):
+    print("Modify:")
+    if not leaf_page['modify']:
+        print("No modifies")
+        return
+
+    leaf_modify = leaf_page['modify'].dereference()
+    #dbg('modify', leaf_modify)
+    row_leaf_insert = leaf_modify['u2']['row_leaf']['insert']
+    #dbg('row store', row_leaf_modify)
+    if not row_leaf_insert:
+        print("No insert list")
+    else:
+        print("Insert list:")
+        dump_skip_list(row_leaf_insert.dereference().dereference())
+    
+    row_leaf_update = leaf_modify['u2']['row_leaf']['update']
+    if not row_leaf_update:
+        print("No update list")
+    else:
+        print("Update list:")
+        leaf_num_entries = int(leaf_page['entries'])
+        for i in range(0, leaf_num_entries):
+            dump_update_chain(row_leaf_update[i])
+
+def dump_disk(leaf_page):
+    leaf_num_entries = int(leaf_page['entries'])
+    dbg('in-memory page:', leaf_page)
+    dsk = leaf_page['dsk'].dereference()
+    if int(dsk.address) == 0:
+        print("No page loaded from disk.")
+        return
+    dbg('on-disk page:', dsk)
+    wt_page_header_size = 28
+    wt_block_header_size = 12
+    page_bytes = gdb.selected_inferior().read_memory(int(dsk.address) + wt_page_header_size + wt_block_header_size, int(dsk['mem_size'])).tobytes()
+    print("Dsk:\n" + str(page_bytes))
+
+def dump_handle(dhandle):
+    print("Dumping: " + dhandle['name'].string())
+    btree = get_btree_handle(dhandle)
+    root = btree['root']
+    root_page = root['page'].dereference()
+    #dbg('btree', get_btree_handle(user))
```

Comment:
Can we pass some arguments to enable these debug prints?

Reply:
I thought about this and just turned them on in the end. It's easier for someone to disable something that's too noisy than to remember to enable it.

I think this is very much a starting point. I'd like to improve both how these are loaded and the consistency of naming/interface. Hopefully whats here now will be a reasonable starting point for that.